### PR TITLE
Let `FallbackInitialHandler` extend `DummyInitialHandler` to using original methods.

### PR DESCRIPTION
**Pull request by FallenCrystal**
Tested the random uuid (FIXED mode) of the JPremium plugin. It works.
However, it is uncertain whether there will be any other impacts.
We may need to do more testing.

I don't know what to call them yet. And the design seems a bit bad now.

Protocolize Tests:
  - [x] Ping (async ping process may cause some unexcepted behavior. if the connection is closed while processing)
  - [x] Join proxy server
  - [x] Join backend server
  - [x] Custom packet inject & listener

Authorization plugin test (with random uniqueId):
  - [x] JPremium
  - [x] LibreLogin

TL;DR before it was fully tested. don't merge the pull request. changes that break compatibility may be introduced.

### CODE REVIEW

**Comment by FallenCrystal** [comment]
Should we set up the original InitialHandler when verified?
I think it might be a little bit better. But I'm not sure if this creates any anomalies.

File: sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/FallbackListener.java [hunk 1]
```diff
@@ -42,6 +43,9 @@ public void handle(final @NotNull LoginEvent event) {
     // the connecting player is greater than the configured amount
     final int maxOnlinePerIp = Sonar.get().getConfig().getMaxOnlinePerIp();
 
+    // Not FallbackInitialHandler. It just original InitialHandler.
+    final InitialHandler original = (InitialHandler) event.getConnection();
+    final FallbackInitialHandler initialHandler = Objects.requireNonNull(FallbackInitialHandler.recordMaps.get(original));
```
Comment (FallenCrystal) [inline sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/FallbackListener.java]: I'm looking for a way to replace it. or make the required method a static method.

**Comment by FallenCrystal** [comment]
We may no longer need the dedicated set/get uniqueId method. It should never reach the fallback handler.

**Comment by FallenCrystal** [comment]
I think that's enough. However, there may be some issues with naming conventions.

**Comment by jonesdevelopment** [comment]
I like this approach, however, did you check if the constant reflection on the fields causes performance issues?

File: sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/handler/DummyInitialHandler.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
+package xyz.jonesdev.sonar.bungee.fallback.handler;
```
Comment (jonesdevelopment) [inline sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/handler/DummyInitialHandler.java:18]: Please use the same license header as the other files in Sonar:

```
/*
 * Copyright (C) 2023-2024 Sonar Contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
```

File: sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/handler/ReplaceInitialHandler.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package xyz.jonesdev.sonar.bungee.fallback.handler;
```
Comment (jonesdevelopment) [inline sonar-bungee/src/main/java/xyz/jonesdev/sonar/bungee/fallback/handler/ReplaceInitialHandler.java:19]: Please use the same license header as the other files in Sonar (read previous comment).

**Comment by jonesdevelopment** [comment]
I'll test it out and see if we can maybe simplify some of it since I think there may be an idea to implement the same functionality with less code/classes.

**Comment by FallenCrystal** [comment]
> I like this approach, however, did you check if the constant reflection on the fields causes performance issues?

The fields in FallbackInitialHandler are useless. will be removed in future commits.

for listener, `LoginEvent` only fires when verified. I don't think it's a serious problem. (unless the server needs to withstand a large number of verified connections at the same time)

**Comment by jonesdevelopment** [comment]
> for listener, `LoginEvent` only fires when verified. I don't think it's a serious problem. (unless the server needs to withstand a large number of verified connections at the same time)

That's intended since the player is disconnected after verification and has to reconnect.

**Comment by FallenCrystal** [comment]
> That's intended since the player is disconnected after verification and has to reconnect.

but now it is the original InitialHandler that is offered. idk of any other way to get ChannelWrapper. or create a map to get it.

**Comment by jonesdevelopment** [comment]
I just think that there has to be a different way of offering the same level of compatibility (this is likely related to #171) as I don't think creating multiple classes and an entire hierarchy just for listening to a single packet is a bit overkill, you know?

**Comment by FallenCrystal** [comment]
but I really don't know of any way to fix these issues (with initialhandler).maybe we can consider using a separate ChannelDuplexHandler.

> i mean if u no longer need to handle pings async. i'll choose a separate handler. this also makes it possible to solve the problems caused by replacing the InitialHandler.

**Comment by jonesdevelopment** [comment]
Can't we just use the `FallbackHandlerBoss` to listen in `channelRead(...)`? If it's a Login packet we can just perform the stuff we would normally perform in the InitialHandler. Then, no replacement must take place.

**Comment by FallenCrystal** [comment]
> Can't we just use the `FallbackHandlerBoss` to listen in `channelRead(...)`? If it's a Login packet we can just perform the stuff we would normally perform in the InitialHandler. Then, no replacement must take place.

i didn't think about it.  you are right.

**Comment by jonesdevelopment** [comment]
Anyway, do you know anything about #172?

#171 should be fixed when this PR/discussion is resolved.

**Comment by FallenCrystal** [comment]
> Anyway, do you know anything about #172?

yes. i haven't even used geyser.

**Comment by FallenCrystal** [comment]
if i migrate to HandlerBoss now, should i open another pr? or just do this on this branches.

**Comment by jonesdevelopment** [comment]
> if i migrate to HandlerBoss now, should i open another pr? or just do this on this branches.

I think it makes more sense to create a new branch since you'd have to revert all your changes which isn't a good idea in my opinion.
